packages/medusa/src/services/__tests__/customer.js

```diff
@@ -246,9 +253,52 @@ describe("CustomerService", () => {
     })
   })
 
+  describe("update customer groups", () => {
+    const customerRepository = MockRepository({
+      findOne: (query) => {
+        return Promise.resolve({ id: IdMap.getId("ironman") })
+      },
+    })
+
+    const addressRepository = MockRepository({
+      create: (data) => data,
+      save: (data) => Promise.resolve(data),
+    })
+
+    const customerGroupRepository = MockRepository({
+      findByIds: jest.fn().mockImplementation(() => Promise.resolve()),
+    })
+
+    const customerService = new CustomerService({
+      manager: MockManager,
+      addressRepository,
+      customerRepository,
+      customerGroupRepository,
+      customerGroupService,
+      eventBusService,
+    })
+
+    beforeEach(async () => {
+      jest.clearAllMocks()
+    })
+
+    it("calls `customerGroupService.list` if `groups` prop is received as a param", async () => {
+      await customerService.update(IdMap.getId("ironman"), {
+        groups: [{ id: "group-id", name: "group-name" }],
```

Comment:
`name` should technically not be passed - these tests will probably fail once you have added `ValidateNested` so won't highlight the rest :) 
 
```suggestion
        groups: [{ id: "group-id" }],
```